src/main/java/io/aiven/kafka/auth/json/AivenAcl.java

```diff
@@ -38,26 +38,34 @@ public class AivenAcl {
     @SerializedName("resource_pattern")
     public final String resourceRePattern;
 
+    @SerializedName("permission_type")
+    private final AclPermissionType permissionType;
```

Comment:
I think `public final` should be fine and the check from `getPermissionType` could be done in the constructor (for uniformity)

Reply:
Maybe I'm mistaken: the constructor isn't called by the JSON library.

Reply:
That's right, the constructor is not called: see here https://github.com/aiven/auth-for-apache-kafka/pull/135#discussion_r1258451405